### PR TITLE
 Combine Title and Overview for multi-episodes files for the TMDB provider

### LINKED ISSUE #2980: Download all titles for multi-episodes
I have several TV series that contain multiple episodes per file, usually 2, sometimes more. The naming convention I'm using for the files is "SXXEXX-EXX Title 1 ~ Title 2" which seems to work for TVDB scraping. But in jellyfin, the metadata is only downloaded for the first episode. I'd like to be able to see the title/description for each episode in the file. At the very least, download just the description/rating/thumbnail for the first episode, but show the titles of all episodes in the file. That way I can search my library for any episode in the series.

### PR DESCRIPTION
**Pull request by scampower3**
Now multi-episodes file show all episodes titles and overviews instead of only just the first episode's title and overview.

**Changes**

- Checks if there is an IndexNumberEnd. If it exist, do multiple request for all episodes between IndexNumber and IndexNumber end. Concats the name and overviews with a " / " delimiting each other.

**Issues**
Partially fixes  #2980

### CODE REVIEW

**Review by Bond-009 — CHANGES_REQUESTED** [review]

**Review by JPVenson — APPROVED** [review]

**Review by 1337joe — APPROVED** [review]

File: MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs [hunk 1]
```diff
@@ -102,9 +104,48 @@ public async Task<MetadataResult<Episode>> GetMetadata(EpisodeInfo info, Cancell
                 return metadataResult;
             }
 
-            var episodeResult = await _tmdbClientManager
-                .GetEpisodeAsync(seriesTmdbId, seasonNumber.Value, episodeNumber.Value, info.SeriesDisplayOrder, info.MetadataLanguage, TmdbUtils.GetImageLanguagesParam(info.MetadataLanguage), cancellationToken)
-                .ConfigureAwait(false);
+            var episodeResult = new TvEpisode();
```
Comment (Bond-009) [inline MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs:107]: This value never gets used?

File: MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs [hunk 2]
```diff
@@ -102,9 +104,48 @@ public async Task<MetadataResult<Episode>> GetMetadata(EpisodeInfo info, Cancell
...
+
+                if (result.Count > 0)
+                {
+                    episodeResult = result[0];
+                }
+                else
+                {
+                    return metadataResult;
+                }
+
```
Comment (Bond-009) [inline MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs:136]: Should we add a fast return if only 1 result was found?

File: MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs [hunk 3]
```diff
@@ -102,9 +104,48 @@ public async Task<MetadataResult<Episode>> GetMetadata(EpisodeInfo info, Cancell
...
+                    var episodeInfo = await _tmdbClientManager.GetEpisodeAsync(seriesTmdbId, seasonNumber.Value, episode.Value, info.SeriesDisplayOrder, info.MetadataLanguage, TmdbUtils.GetImageLanguagesParam(info.MetadataLanguage), cancellationToken).ConfigureAwait(false);
+                    if (episodeInfo is not null)
+                    {
+                        result.Add(episodeInfo);
+                    }
+                }
+
+                if (result.Count > 0)
+                {
+                    episodeResult = result[0];
```
Comment (Bond-009) [inline MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs:130]:
```suggestion
                List<TvEpisode>? result = null;
                for (int? episode = startindex; episode <= endindex; episode++)
                {
                    var episodeInfo = await _tmdbClientManager.GetEpisodeAsync(seriesTmdbId, seasonNumber.Value, episode.Value, info.SeriesDisplayOrder, info.MetadataLanguage, TmdbUtils.GetImageLanguagesParam(info.MetadataLanguage), cancellationToken).ConfigureAwait(false);
                    if (episodeInfo is not null)
                    {
                        (result =?? new List<TvEpisode>()).Add(episodeInfo);
                    }
                }

                if (result is not null)
                {
                    episodeResult = result[0];
```

File: MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs [hunk 4]
```diff
@@ -102,9 +104,48 @@ public async Task<MetadataResult<Episode>> GetMetadata(EpisodeInfo info, Cancell
                 return metadataResult;
             }
 
-            var episodeResult = await _tmdbClientManager
-                .GetEpisodeAsync(seriesTmdbId, seasonNumber.Value, episodeNumber.Value, info.SeriesDisplayOrder, info.MetadataLanguage, TmdbUtils.GetImageLanguagesParam(info.MetadataLanguage), cancellationToken)
-                .ConfigureAwait(false);
+            var episodeResult = new TvEpisode();
+            if (!info.IndexNumberEnd.HasValue)
```
Comment (Bond-009) [inline MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs:108]: Change these branches around, i.e multi episodes first with `if (info.IndexNumberEnd.HasValue)`

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs (hunk 2 above)
Comment (scampower3) [reply MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs:136]: I can shift the following code 

                var name = new StringBuilder(episodeResult.Name);
                var overview = new StringBuilder(episodeResult.Overview);

                for (int i = 1; i < result.Count; i++)
                {
                    name.Append(" / " + result[i].Name);
                    overview.Append(" / " + result[i].Overview);
                }

                episodeResult.Name = name.ToString();
                episodeResult.Overview = overview.ToString();

to be under

                if (result is not null)

and do a check if there is only one result. Then, can exit the if statment ealier without going through the for loop?

File: MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs (hunk 3 above)
Comment (scampower3) [reply MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs]: Added

File: MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs (hunk 4 above)
Comment (scampower3) [reply MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs]: Swapped

File: MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs (hunk 1 above)
Comment (scampower3) [reply MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs]: Required for this section of code to work

https://github.com/scampower3/jellyfin/blob/477d74d5c7be3c42bd5859134885d0e5afe3af30/MediaBrowser.Providers/Plugins/Tmdb/TV/TmdbEpisodeProvider.cs#L155C1-L256C40

**Review by Bond-009 — APPROVED** [review]

**Comment by SmearyThrower** [comment]
is this part of jellyfin's tmdb implementation since rn for me combination of titles only works on the tvdb

**Comment by nielsvanvelzen** [comment]
This pull request is part of the current 10.9 builds.

**Comment by SmearyThrower** [comment]
is this feature also implemented for tv maze?

**Pull request merged**